otx/core/action_dataset_adapter.py

```diff
@@ -0,0 +1,169 @@
+"""Action Base / Classification / Detection Dataset Adapter."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# pylint: disable=invalid-name, too-many-locals, no-member
+import os
+import os.path as osp
+from typing import List
+
+from datumaro.components.dataset import Dataset as DatumaroDataset
+from datumaro.components.annotation import AnnotationType as DatumaroAnnotationType
+from datumaro.components.annotation import Bbox as DatumaroBbox
+from datumaro.components.annotation import Label as DatumaroLabel
+
+from otx.core.base_dataset_adapter import BaseDatasetAdapter
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.subset import Subset
+from otx.api.entities.id import ID
+from otx.api.entities.image import Image
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.rectangle import Rectangle
+from otx.api.entities.label import LabelEntity
+from otx.api.entities.annotation import (Annotation, AnnotationSceneEntity, AnnotationSceneKind, NullAnnotationSceneEntity)
+from otx.utils.logger import get_logger
+
+logger = get_logger()
+
+class ActionBaseDatasetAdapter(BaseDatasetAdapter):
+    def import_dataset(
+        self,
+        train_data_roots: str,
+        val_data_roots: str,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None
+    ) -> DatumaroDataset:
+        """ Import multiple videos that have CVAT format annotation. """
+        self.dataset = {}
+        self.dataset[Subset.TRAINING] = self._prepare_cvat_pair_data(train_data_roots)
+        self.dataset[Subset.VALIDATION] = self._prepare_cvat_pair_data(val_data_roots)
+        if test_data_roots:
+            self.dataset[Subset.TESTING] = self._prepare_cvat_pair_data(test_data_roots)
```

Comment:
has it been decided to use cvat video format in Geti together?

Reply:
Not decided yet, CVAT is just an example for the first step.